### ISSUE TITLE
post-quantum algorithms

### ISSUE THREAD

**Issue by gl-donald** [tmcqueen-materials/kafkacrypto#2]
Hello, I'm curious on why you decided to implement Curve25519+sntrup761 as the post quantum algorithm?
Do you have any idea if the algorithms just recommended by NIST could be implemented (CRYSTALS-Kyber, CRYSTALS-Dilithium, Sphincs+ and FALCON)?

Great work
Thanks

**Comment by tmcqueen-materials** [comment]
@gl-donald : thanks for the question! It was implemented because we wanted PQ crypto employed to protect against collect-now-decrypt-later attacks faster than NIST was making its decisions. So we went with the same choice as [OpenSSH did](https://www.openssh.com/txt/release-8.5) in the interim.

We plan to add NIST-standardized PQ key exchange and PQ signature support at the same time, once the landscape of that settles down a bit more (possibly not until the 4th round standardization is complete).
